File: _p1_problems/Remove Duplicates from Sorted Array/Solution.java
public class Solution {
    public int removeDuplicates(int[] A) {
        if (A.length == 0) return 0;
        int i = 0;
        for (int j = 0; j < A.length; j++) {
            if (A[i] != A[j]) {
                i++;
                A[i] = A[j];
            }
        }
        return i + 1;
    }
}

File: _p1_problems/Remove Duplicates from Sorted Array/Solution.rb
class Solution
  def removeDuplicates(arr)
    return 0 if arr.nil? or arr.empty?

    len = 1
    (1...arr.length).each do |i|
      if arr[len - 1] != arr[i]
        arr[len] = arr[i]
        len += 1
      end
    end
    len
  end
end
